mesa_frames/abstract/space.py

```diff
@@ -733,3 +738,419 @@ def full_cells(self) -> DataFrame:
         return self._sample_cells(
             None, with_replacement=False, condition=self._full_cell_condition
         )
+
+
+class GridDF(DiscreteSpaceDF):
+    """The GridDF class is an abstract class that defines the interface for all grid classes in mesa-frames.
+    Inherits from DiscreteSpaceDF.
+
+    Methods
+    -------
+    __init__(model: 'ModelDF', dimensions: Sequence[int], torus: bool = False, capacity: int | None = None, neighborhood_type: str = 'moore')
+        Create a new GridDF object.
+    out_of_bounds(pos: GridCoordinate | GridCoordinates) -> DataFrame
+        Check whether the input positions are out of bounds in a non-toroidal grid.
+
+    Properties
+    ----------
+    dimensions : Sequence[int]
+        The dimensions of the grid
+    neighborhood_type : Literal['moore', 'von_neumann', 'hexagonal']
+        The type of neighborhood to consider
+    torus : bool
+        If the grid is a torus
+    """
+
+    _grid_capacity: (
+        GridCapacity  # Storing the remaining capacity of the cells in the grid
+    )
+    _neighborhood_type: Literal[
+        "moore", "von_neumann", "hexagonal"
+    ]  # The type of neighborhood to consider
+    _offsets: DataFrame  # The offsets to compute the neighborhood of a cell
+    _torus: bool  # If the grid is a torus
+
+    def __init__(
+        self,
+        model: "ModelDF",
+        dimensions: Sequence[int],
+        torus: bool = False,
+        capacity: int | None = None,
+        neighborhood_type: str = "moore",
+    ):
+        """Grid cells are indexed, where [0, ..., 0] is assumed to be the
+        bottom-left and [dimensions[0]-1, ..., dimensions[n]-1] is the top-right. If a grid is
```

Comment:
@tpike3 and I had once accidentally loaded a sugarscape map in a way that is the reverse of the convention here. A NumPy 2D array is printed such that the last element is displayed at a bottom right. If you view https://github.com/projectmesa/mesa-examples/blob/main/examples/sugarscape_g1mt/sugarscape_g1mt/sugar-map.txt, the last element is also at bottom right. We have to add a caution sentence about this somehow.

Reply:
I know that you choose the convention of the last element being top right because that's the way it is done with the Cartesian coordinates, and also Matplotlib/Seaborn plot outputs. I agree with it.

Reply:
The one consistent with Cartesian coordinates is better. I think the documentation shouldn't be placed in the `__init__` of `GridDF`, but instead in the equivalent of `np.genfromtxt`: https://github.com/projectmesa/mesa-examples/blob/e137a60e4e2f2546901bec497e79c4a7b0cc69bb/examples/sugarscape_g1mt/sugarscape_g1mt/model.py#L93-L94.

Reply:
To clarify: by "the documentation", I meant specifically about the warning that the convention is *different* from `np.genfromtxt`.

Reply:
I don't think there is an equivalent of `np.genfromtxt` right now (should it be added?).
I will add the warning to the class docstring for now.